source/game_sa/ClothesBuilder.cpp

```diff
@@ -75,34 +91,169 @@ void CClothesBuilder::ReleaseGeometry(int32 numToRelease) {
         --i;
 }
 
-// 0x5A47E0
-void FindAtomicFromNameCB(RpAtomic* atomic, void* data) {
-    plugin::Call<0x5A47E0, RpAtomic*, void*>(atomic, data);
-}
-
 // 0x5A4810
-void GetAtomicWithName(RpClump* clump, const char* name) {
-    plugin::Call<0x5A4810, RpClump*, const char*>(clump, name);
+RpAtomic* GetAtomicWithName(RpClump* clump, const char* name) {
+    struct Context {
+        notsa::ci_string_view name{};
+        RpAtomic*             a{};
+    } c{name};
+    RpClumpForAllAtomics(clump, [](RpAtomic* a, void* data) { // 0x5A47E0
+        auto& c = *static_cast<Context*>(data);
+
+        if (c.name == GetFrameNodeName(RpAtomicGetFrame(a))) {
+            c.a = a;
+        }
+
+        return a;
+    }, &c);
+    return c.a;
 }
 
 // 0x5A4840
-void CClothesBuilder::sub_5A4840() {
-
+void CClothesBuilder::AddWeightToBoneVertex(float (&weights)[8], uint8(&boneVertexIdxs)[8], float weightToAdd, RwUInt32 targetVertexIdx) { // Unknown OG name
+    if (weightToAdd == 0.f) {
+        return;
+    }
+    for (auto i = 0; i < 8; i++) {
+        if (weights[i] == 0.f) { // Weight not yet used?
+            boneVertexIdxs[i] = targetVertexIdx; // Add to list
+            weights[i]        = weightToAdd;
+            weights[i + 1]    = 0.f;             // Mark next as unused [Though this step in our case is not necessary as the whole weights array is already zero-inited]
+            return;
+        }
+        if (boneVertexIdxs[i] == targetVertexIdx) { // Already in the list, use that
+            weights[i] += weightToAdd;
+            return;
+        }
+    }
+    NOTSA_UNREACHABLE(); // OG code had UB in this case
 }
 
 // 0x5A48B0
-void CClothesBuilder::StoreBoneArray(RpClump* clump, int32 a2) {
-    plugin::Call<0x5A48B0, RpClump*, int32>(clump, a2);
+void CClothesBuilder::StoreBoneArray(RpClump* clump, int32 idx) {
+    const auto a = GetAtomicWithName(clump, "normal");
+    assert(a);
+
+    const auto h = RpSkinAtomicGetHAnimHierarchy(a);
+    assert(h);
+    
+    rng::fill(gBoneIndices[idx], -1);
+    for (auto i = h->numNodes; i-- > 0;) {
+        gBoneIndices[idx][i] = static_cast<int16>(h->pNodeInfo[i].nodeID);
+    }
```

Comment:
```suggestion
    for (auto i = h->numNodes; i-- > 0;) {
        gBoneIndices[idx][i] = static_cast<int16>(h->pNodeInfo[i].nodeID);
    }
    if(h->numNodes < 64) gBoneIndices[idx][h->numNodes] = -1;
```

I think there will be less action this way.

Reply:
it's not the same behavior

Reply:
> it's not the same behavior

Но этого достаточно, если учитывать следующее: https://github.com/gta-reversed/gta-reversed-modern/pull/587/files#diff-91b69700e5f11144464257909a763395ef429f5b84fcf5381d9bb2ca4d7ab811R299

Reply:
i see. if all of xref's to it is reversed then that's fine i guess

Reply:
Maybe write it as FIX_BUGS?)

Reply:
Если отреверсили и везде массив используется также, как и в этом примере (https://github.com/gta-reversed/gta-reversed-modern/pull/587/files#diff-91b69700e5f11144464257909a763395ef429f5b84fcf5381d9bb2ca4d7ab811R299-R301)